spec/lib/whitehall_importer/create_file_attachment_revision_spec.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+RSpec.describe WhitehallImporter::CreateFileAttachmentRevision do
+  let(:whitehall_file_attachment) do
+    build(:whitehall_export_file_attachment)
+  end
+
+  context "creates a file attachment" do
+    it "fetches file from asset-manager" do
+      create_revision = described_class.new(whitehall_file_attachment)
+      expect(create_revision.call).to have_requested(:get, whitehall_file_attachment["url"])
+    end
+
+    it "creates a FileAttachment::BlobRevision" do
+      expect { described_class.call(whitehall_file_attachment) }
+        .to change { FileAttachment::BlobRevision.count }.by(1)
+    end
+
+    it "creates a FileAttachment::Revision" do
+      expect { described_class.call(whitehall_file_attachment) }
+        .to change { FileAttachment::Revision.count }.by(1)
+    end
+
+    it "creates a FileAttachment::MetadataRevision" do
+      metadata_revision = nil
+      expect { metadata_revision = described_class.call(whitehall_file_attachment) }
+        .to change { FileAttachment::MetadataRevision.count }.by(1)
+
+      expect(metadata_revision.title).to eq(whitehall_file_attachment["title"])
+    end
+
+    it "creates a FileAttachment::Asset" do
+      expect { described_class.call(whitehall_file_attachment) }
+        .to change { FileAttachment::Asset.count }.by(1)
+    end
```

Comment:
We can probably test all these with a single test as per: https://github.com/alphagov/content-publisher/blob/920864a5e072918fe2475f1ab5340b2fde8c47d9/spec/lib/whitehall_importer/create_image_revision_spec.rb#L6-L14

From a quick glance I think this test could use being a bit more consistent with that one.

Reply:
Sure, I've merged the tests where it's testing for metadata and blob record being created into whether the revision has been created as a file_attachment_revision must have metadata and blob record to of been created!